app/code/Mastering/SampleModule/Model/ItemRepository.php

```diff
@@ -10,21 +10,148 @@
+
     private $collectionFactory;
 
-    public function __construct(CollectionFactory $collectionFactory)
+
+    /**
+     * @var Data\BlockSearchResultsInterfaceFactory
+     */
+    protected $searchResultsFactory;
+
+
+    /**
+     * @var DataObjectHelper
+     */
+    protected $dataObjectHelper;
+
+    /**
+     * @var DataObjectProcessor
+     */
+    protected $dataObjectProcessor;
+
+    /**
+     * @var \Mastering\SampleModule\Api\Data\TestInterfaceFactory
+     */
+    protected $dataitemFactory;
+
+
+    public function __construct(
+        ResourceItem $resource,
+        ItemFactory $itemFactory,
+        CollectionFactory $collectionFactory,
+        \Mastering\SampleModule\Api\Data\ItemInterfaceFactory $dataitemFactory,
+        \Mastering\SampleModule\Api\Data\ItemSearchResultsInterfaceFactory $searchResultsFactory,
+        DataObjectHelper $dataObjectHelper,
+        DataObjectProcessor $dataObjectProcessor
+    )
     {
+        $this->resource = $resource;
         $this->collectionFactory = $collectionFactory;
+        $this->itemFactory = $itemFactory ;
+        $this->dataitemFactory = $dataitemFactory;
+        $this->searchResultsFactory = $searchResultsFactory;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->dataObjectProcessor = $dataObjectProcessor;
     }
 
-
-    public function getList()
+   /* public function getList()
     {
         return $this->collectionFactory->create()->getItems();
+    }*/
+
+    /**
+     * Load Test data collection by given search criteria
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $criteria
+     * @return \Mastering\SampleModule\Model\ResourceModel\Item\Collection
+     */
+    public function getList(\Magento\Framework\Api\SearchCriteriaInterface $criteria)
+    {
+        $collection = $this->collectionFactory->create();
+        foreach ($criteria->getFilterGroups() as $filterGroup) {
+            foreach ($filterGroup->getFilters() as $filter) {
+                if ($filter->getField() === 'id') {
+                    $collection->addFilter($filter->getValue(), false);
+                    continue;
+                }
+                $condition = $filter->getConditionType() ?: 'eq';
+                $collection->addFieldToFilter($filter->getField(), [$condition => $filter->getValue()]);
+            }
+        }
+
+        $sortOrders = $criteria->getSortOrders();
+        if ($sortOrders) {
+            /** @var SortOrder $sortOrder */
+            foreach ($sortOrders as $sortOrder) {
+                $collection->addOrder(
+                    $sortOrder->getField(),
+                    ($sortOrder->getDirection() == SortOrder::SORT_ASC) ? 'ASC' : 'DESC'
+                );
+            }
+        }
+        $collection->setCurPage($criteria->getCurrentPage());
+        $collection->setPageSize($criteria->getPageSize());
+
```

Comment:
just add in class usage the collection processor : use Magento\Framework\Api\SearchCriteria\CollectionProcessor;
and now use it like this here:
 $this->collectionProcessor->process($criteria, $collection);
then continue from $searchResults = ..